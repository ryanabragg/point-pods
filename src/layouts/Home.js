import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import { withAPI } from '../api';

import Button from '@material-ui/core/Button';

import AddIcon from '@material-ui/icons/Add';

import AppMenu from '../components/AppMenu';
import TournamentCards from '../components/TournamentCards';

const styles = theme => ({
  actionButton: {
    margin: 2 * theme.spacing.unit,
    position: 'absolute',
    bottom: 0,
    right: 0,
  },
});

class Home extends Component {
  state = {
    tournaments: []
  };

  componentDidMount() {
    this.props.api.Tournaments.all()
      .then(list => this.setState({ tournaments: list }))
      .catch(error => this.props.notification(error.message, 'error'));
  }

  render() {
    const { classes } = this.props;
    const { tournaments } = this.state;
    return (
      <div>
        <AppMenu title='Point Pods' />
        <Button
          variant='fab'
          color='secondary'
          aria-label='New Tournament'
          className={classes.actionButton}
          href='/new'
        >
          <AddIcon />
        </Button>
        <TournamentCards tournaments={tournaments} status='Incomplete' />
      </div>
    );
  }
}

Home.propTypes = {
  api: PropTypes.object.isRequired, // added by withAPI
  classes: PropTypes.object.isRequired, // added by withStyles
};

export default withStyles(styles, { withTheme: true })(withAPI(Home));
