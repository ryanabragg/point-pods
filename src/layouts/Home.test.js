import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Home from './Home';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Home />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('matches the prior snapshot', () => {
  const tree = renderer.create(<Home />);
  expect(tree.toJSON()).toMatchSnapshot();
});
