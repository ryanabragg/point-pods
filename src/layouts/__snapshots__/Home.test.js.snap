// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the prior snapshot 1`] = `
<div>
  <div
    className="AppMenu-root-144"
  >
    <header
      className="MuiPaper-root-159 MuiPaper-elevation4-165 MuiAppBar-root-151 MuiAppBar-positionFixed-152 MuiAppBar-colorPrimary-157 mui-fixed"
    >
      <div
        className="MuiToolbar-root-186 MuiToolbar-regular-188 MuiToolbar-gutters-187 AppMenu-nowrap-146"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root-196 MuiIconButton-root-190 MuiIconButton-colorInherit-191 AppMenu-menuButton-147"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-195"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-199"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-206"
          />
        </button>
        <h2
          className="MuiTypography-root-213 MuiTypography-title-219 MuiTypography-colorInherit-232 AppMenu-flex-145"
        >
          Point Pods
        </h2>
      </div>
    </header>
    <div
      className="AppMenu-spacer-149"
    />
  </div>
  <a
    aria-label="New Tournament"
    className="MuiButtonBase-root-196 MuiButton-root-251 MuiButton-fab-266 MuiButton-contained-260 MuiButton-containedSecondary-262 MuiButton-raised-263 MuiButton-raisedSecondary-265 APIComponent-actionButton-143"
    href="/new"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-252"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-199"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-206"
    />
  </a>
  <div
    className="TournamentCards-root-275"
  >
    <div
      className="MuiPaper-root-159 MuiPaper-elevation1-162 MuiPaper-rounded-160 MuiCard-root-278 TournamentCards-card-276"
      id="not-found"
    >
      <div
        className="MuiCardHeader-root-279"
      >
        <div
          className="MuiCardHeader-content-282"
        >
          <span
            className="MuiTypography-root-213 MuiTypography-headline-218 MuiCardHeader-title-283"
          >
            No Tournaments Found
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches the prior snapshot 2`] = `
<div>
  <div
    className="AppMenu-root-144"
  >
    <header
      className="MuiPaper-root-159 MuiPaper-elevation4-165 MuiAppBar-root-151 MuiAppBar-positionFixed-152 MuiAppBar-colorPrimary-157 mui-fixed"
    >
      <div
        className="MuiToolbar-root-186 MuiToolbar-regular-188 MuiToolbar-gutters-187 AppMenu-nowrap-146"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root-196 MuiIconButton-root-190 MuiIconButton-colorInherit-191 AppMenu-menuButton-147"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-195"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-199"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-206"
          />
        </button>
        <h2
          className="MuiTypography-root-213 MuiTypography-title-219 MuiTypography-colorInherit-232 AppMenu-flex-145"
        >
          Point Pods
        </h2>
      </div>
    </header>
    <div
      className="AppMenu-spacer-149"
    />
  </div>
  <a
    aria-label="New Tournament"
    className="MuiButtonBase-root-196 MuiButton-root-251 MuiButton-fab-266 MuiButton-contained-260 MuiButton-containedSecondary-262 MuiButton-raised-263 MuiButton-raisedSecondary-265 APIComponent-actionButton-143"
    href="/new"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-252"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-199"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-206"
    />
  </a>
  <div
    className="TournamentCards-root-275"
  >
    <div
      className="MuiPaper-root-159 MuiPaper-elevation1-162 MuiPaper-rounded-160 MuiCard-root-278 TournamentCards-card-276"
      id="mock0"
    >
      <div
        className="MuiCardHeader-root-279"
      >
        <div
          className="MuiCardHeader-content-282"
        >
          <span
            className="MuiTypography-root-213 MuiTypography-headline-218 MuiCardHeader-title-283"
          >
            test0
          </span>
          <span
            className="MuiTypography-root-213 MuiTypography-body1-222 MuiTypography-colorTextSecondary-236 MuiCardHeader-subheader-284"
          >
            cat0
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-285"
      >
        <p
          className="MuiTypography-root-213 MuiTypography-body1-222"
        >
          4 Total Players
        </p>
        <p
          className="MuiTypography-root-213 MuiTypography-body1-222"
        >
          3 Active Players
        </p>
      </div>
      <div
        className="MuiCardActions-root-286"
      >
        <a
          className="MuiButtonBase-root-196 MuiButton-root-251 MuiButton-text-253 MuiButton-flat-256 MuiButton-sizeSmall-272 TournamentCards-right-277 MuiCardActions-action-287"
          href="/new/mock0"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-252"
          >
            Continue
          </span>
          <span
            className="MuiTouchRipple-root-206"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-159 MuiPaper-elevation1-162 MuiPaper-rounded-160 MuiCard-root-278 TournamentCards-card-276"
      id="mock2"
    >
      <div
        className="MuiCardHeader-root-279"
      >
        <div
          className="MuiCardHeader-content-282"
        >
          <span
            className="MuiTypography-root-213 MuiTypography-headline-218 MuiCardHeader-title-283"
          >
            test2
          </span>
          <span
            className="MuiTypography-root-213 MuiTypography-body1-222 MuiTypography-colorTextSecondary-236 MuiCardHeader-subheader-284"
          >
            cat0
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-285"
      >
        <p
          className="MuiTypography-root-213 MuiTypography-body1-222"
        >
          4 Total Players
        </p>
        <p
          className="MuiTypography-root-213 MuiTypography-body1-222"
        >
          4 Active Players
        </p>
      </div>
      <div
        className="MuiCardActions-root-286"
      >
        <a
          className="MuiButtonBase-root-196 MuiButton-root-251 MuiButton-text-253 MuiButton-flat-256 MuiButton-sizeSmall-272 TournamentCards-right-277 MuiCardActions-action-287"
          href="/tournament/mock2"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-252"
          >
            Continue
          </span>
          <span
            className="MuiTouchRipple-root-206"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
